File: Palestra di Dennis Ritchie (IMP)/Team 2/Pokemon Battle/Pokemon_battle.py
class pokemon():
    def __init__(self, name, hp, attack, defense):
        self.name = name
        self.hp = hp
        self.attack = attack
        self.defense = defense

    def attack_enemy(self, enemy):
        damage = self.attack - enemy.defense
        enemy.hp -= damage


def battle(pokemon1,pokemon2):
    while pokemon1.hp > 0 and pokemon2.hp > 0:
        pokemon1.attack_enemy(pokemon2)
        print(pokemon1.name, " attacca!")
        if pokemon2.hp <= 0:
            return pokemon1.name
        else:
            pokemon2.attack_enemy(pokemon1)
            print(pokemon2.name, " attacca!")
            if pokemon1.hp <= 0:
                return pokemon2.name

pokemon1 = pokemon("Pikachu", 100, 50, 20)
pokemon2 = pokemon("Charizard", 120, 40, 30)
print("ha vinto: ", battle(pokemon1, pokemon2), "!")

File: Palestra di Dennis Ritchie (IMP)/Pokemon Battle/22-PokemonBattle/PokemonBattle.java

public class PokemonBattle {

    public static void battle(Pokemon pokemon1, Pokemon pokemon2) {
        System.out.println("Battaglia tra " + pokemon1.name + " e " + pokemon2.name + " avviata!");

        while (pokemon1.hp > 0 && pokemon2.hp > 0) {
            pokemon1.attack(pokemon2);
            pokemon2.displayHP();


            if (pokemon2.hasFainted()) {
                System.out.println(pokemon2.name + " has perso! " + pokemon1.name + " vince!");
                break;
            }

            pokemon2.attack(pokemon1);
            pokemon1.displayHP();

            if (pokemon1.hasFainted()) {
                System.out.println(pokemon1.name + " has perso! " + pokemon2.name + " vince!");
                break;
            }
        }
    }

    public static void main(String[] args) {
        Pokemon pikachu = new Pokemon("Pikachu", 100, 55, 40);
        Pokemon bulbasaur = new Pokemon("Bulbasaur", 105, 50, 45);

        battle(pikachu, bulbasaur);
    }
}
